src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -113,20 +117,51 @@ typedef int (*set_runtime_t)(halide_malloc_t user_malloc,
 PipelineContext run_context(stack_alignment, stack_size);
 
 int halide_hexagon_remote_initialize_kernels(const unsigned char *code, int codeLen,
+                                             int use_dlopen, int use_dlopenbuf,
                                              handle_t *module_ptr) {
-    elf_t *lib = obj_dlopen_mem(code, codeLen);
-    if (!lib) {
-        log_printf("dlopen failed");
-        return -1;
-    }
+    log_printf("halide_hexagon_remote_initialize_kernels ->\n");
+    void *lib = NULL;
+    elf_t *elib = NULL;
+    if (use_dlopenbuf) {
+        log_printf("dlopenbuf started\n");
+        lib = dlopenbuf( "libhalide_hexagon_host_dlbuf.so", (const char*)code, codeLen, RTLD_LOCAL | RTLD_LAZY);
```

Comment:
Might be worth a comment here saying that the "filename" here is not actually a file?